test/e2e/storage/csimock/csi_honor_pv_reclaim_policy.go

```diff
@@ -81,7 +117,54 @@ var _ = utils.SIGDescribe("CSI Mock honor pv reclaim policy", feature.HonorPVRec
 			gomega.Expect(m.driver.GetCalls(ctx)).To(gomega.ContainElement(gomega.HaveField("Method", gomega.Equal("DeleteVolume"))))
 		})
 
-		ginkgo.It("Dynamic provisioning should honor pv retain reclaim policy", func(ctx context.Context) {
+		ginkgo.It("Dynamic provisioning should honor pv retain reclaim policy when deleting pvc then pv", func(ctx context.Context) {
+			m.init(ctx, testParameters{
+				registerDriver:             true,
+				enableHonorPVReclaimPolicy: true,
+				reclaimPolicy:              ptr.To(v1.PersistentVolumeReclaimRetain),
+			})
+			ginkgo.DeferCleanup(m.cleanup)
+
+			_, pvc := m.createPVC(ctx)
+
+			ginkgo.By(fmt.Sprintf("Waiting for PVC %s to be bound", pvc.Name))
+			pvs, err := e2epv.WaitForPVClaimBoundPhase(ctx, f.ClientSet, []*v1.PersistentVolumeClaim{pvc}, framework.ClaimProvisionTimeout)
+			framework.ExpectNoError(err, "failed to wait for PVC to be bound")
+			gomega.Expect(pvs).To(gomega.HaveLen(1), "expected 1 PV to be bound to PVC, got %d", len(pvs))
+
+			pv := pvs[0]
+			ginkgo.By(fmt.Sprintf("PVC %s is bound to PV %s", pvc.Name, pv.Name))
+			gomega.Expect(pv.Spec.PersistentVolumeReclaimPolicy).To(gomega.Equal(v1.PersistentVolumeReclaimRetain),
+				"expected PV %s to have reclaim policy %s, got %s", pv.Name, v1.PersistentVolumeReclaimRetain, pv.Spec.PersistentVolumeReclaimPolicy)
+
+			ginkgo.By(fmt.Sprintf("Verifying that the PV %s does not have finalizer %s after creation", pv.Name, storagehelpers.PVDeletionProtectionFinalizer))
+			gomega.Consistently(ctx, framework.GetObject(f.ClientSet.CoreV1().PersistentVolumes().Get, pv.Name, metav1.GetOptions{})).
```

Comment:
Can you please shorten the timeout? Now the tests wait for 5 minutes to check that a finalizer does not appear accidentally, which is IMO too long. Would 1 minute be enough?

Reply:
@jsafrane changed it to `ClaimProvisionShortTimeout`. ClaimProvisionShortTimeout is 1 minute.